src/main.rs

```diff
@@ -272,6 +274,57 @@ fn main() {
         }
     };
 
+    let custom_pages_dir = config
+        .directories
+        .custom_pages_dir
+        .as_ref()
+        .map(PathWithSource::path);
+
+    // Note: According to the TLDR client spec, page names must be transparently
+    // lowercased before lookup:
+    // https://github.com/tldr-pages/tldr/blob/main/CLIENT-SPECIFICATION.md#page-names
+    let command = args.command.join("-").to_lowercase();
+
+    if args.edit_patch || args.edit_page {
+        let Some(custom_pages_dir) = custom_pages_dir else {
+            print_error(enable_styles, &anyhow!("Fail to get custom page dir"));
+            process::exit(1);
+        };
+        let _ = create_dir_all(custom_pages_dir);
+
+        let file_name = if args.edit_patch {
+            format!("{command}.patch.md")
+        } else {
+            format!("{command}.page.md")
+        };
+        let custom_page_path = custom_pages_dir.join(file_name);
+        let Ok(editor) = env::var("EDITOR") else {
+            print_error(
+                enable_styles,
+                &anyhow!("`EDITOR` not set. Please set `EDITOR` by `export EDITOR=vim`."),
+            );
+            process::exit(1);
+        };
+        println!("Editing {custom_page_path:?}");
+        match std::process::Command::new(&editor)
+            .arg(custom_page_path.to_str().unwrap())
+            .status()
+        {
+            Ok(status) => {
+                if !status.success() {
+                    print_error(
+                        enable_styles,
+                        &anyhow!("{editor} exit with code {:?}", status.code()),
+                    );
+                }
+            }
+            Err(e) => {
+                print_error(enable_styles, &e.into());
+            }
```

Comment:
In general, we use `print_error` quite often in the added code segment. Maybe it would be better if we made a function that returns `anyhow::Result<()>` and call it from `main`. What do you think?

Reply:
No idea

Reply:
Okay, so let's try it with a function and see which version we like better :)

Reply:
In particular, I think it is a lot more readable if we add a function `spawn_editor(path: &Path) -> anyhow::Result<()>` that creates the directory of `path` and so on, so that we can call it in some way like this from `main`:
```rust
use anyhow::Context;

// ...

fn spawn_editor(path: &Path) -> anyhow::Result<()> {
    // ...
}

fn main() {
    // ...

    if args.edit_patch || args.edit_page {
        let filename = if args.edit_patch {
            format!("{command}.patch.md")
        } else {
            format!("{command}.page.md")
        };

        custom_pages_dir
            .context("To edit custom pages/patches, please specify a custom pages directory.")
            .and_then(|custom_pages_dir| spawn_editor(&custom_pages_dir.join(filename)))
            .unwrap_or_else(|err| {
                print_error(enable_styles, &err);
                process::exit(1);
            });
        return;
    }

    // ...
}
```